source/agora/consensus/validation/Block.d

```diff
@@ -130,6 +132,26 @@ public string isInvalidReason (const ref Block block, Height prev_height,
             return fail_reason;
     }
 
+    Transaction[] expected_txs;
+    foreach(tx; block.txs)
+    {
+        if (tx.type == TxType.Coinbase)
+            continue;
+
+        // Poor mans deep copy
+        Transaction new_tx;
+        new_tx.type = tx.type;
+        new_tx.inputs = tx.inputs.dup;
+        new_tx.outputs = tx.outputs.dup;
+        new_tx.payload = tx.payload;
+        expected_txs ~= new_tx;
+    }
+
+    appendCoinbaseTX(expected_txs, block.header.missing_validators);
+
+    if (!isPermutation(expected_txs, block.txs))
```

Comment:
What does this do?

Reply:
It seems like validation becomes memory intensive now.

Reply:
Well, I am definitely not proud of this part. I'll try to rework it a little.

Reply:
Please check the latest commit, I think it is much better. I will update the docs and squash it, if it gets the OK.